### ISSUE TITLE
__fields_set__ incorrectly set unless using the validate class method

### ISSUE THREAD

**Issue by decherd** [tiangolo/sqlmodel#87]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import SQLModel
from typing import Optional
class HeroUpdate(SQLModel):
    name: Optional[str] = None
    secret_name: Optional[str] = None
    age: Optional[int] = None
hero = HeroUpdate(age=42)
print(hero.dict(exclude_unset=True))
# output = {'name': None, 'secret_name': None, 'age': 42}
```

### Description

*Create a HeroUpdate model
*Create a HeroUpdate instance with unset fields
*Use the .dict attribute with the unset_fields parameter
*unset_fields show up as None rather than not being included in the dictionary
*works fine with Pydantic BaseModel instances

### Operating System

macOS

### SQLModel Version

'0.0.4'

### Python Version

3.9.0

### Additional Context
If it is run as:
hero = HeroUpdate.validate({'age':42})
the \__fields_set\__ are correctly set and it works

**Comment by statt8900** [comment]
I see this as well, and It appears that the main culprit are these lines in SQLModel.__init__:
`        # Do not set values as in Pydantic, pass them through setattr, so SQLAlchemy
        # can handle them
        # object.__setattr__(__pydantic_self__, '__dict__', values)
        object.__setattr__(__pydantic_self__, "__fields_set__", fields_set)
        for key, value in values.items():
            setattr(__pydantic_self__, key, value)`

values is returned from the validated model and contains the default keys and their values. Therefore when setattr is called this adds even the unset default keys to the `__fields_set__` attribute

**Comment by Maffels** [comment]
I have the same problem in my project during development on windows. (SQLModel 0.0.4)

Currently, a workaround might be using the original `pydantic.BaseModel` as the base model for specific update models allows for the use of `exclude_unset` once more. 
Since update models specifically are usually light on inheritance, it's a relatively painless change until the behaviour in `sqlmodel.SQLModel`  is fixed.

**Comment by Iionsroar** [comment]
Is there any update?

Outside FastAPI I tried comparing Pydantic's BaseModel and SQLModel and the exclude_unset parameter really does not do anything to SQLModels

**Comment by tiangolo** [comment]
This was solved by @statt8900 in https://github.com/tiangolo/sqlmodel/pull/117, it will be available in SQLModel `0.0.7`, released in the next hours. :tada:
